.github/workflows/pr-checks.yml

```diff
@@ -0,0 +1,231 @@
+            echo "DEV_PACKAGE_BRANCH=${AZ_TRIMMED_REF}" >> ${GITHUB_ENV}
+          else
+            echo "DEV_PACKAGE_BRANCH=main" >> ${GITHUB_ENV}
+          fi
+      - name: Clone scaffolding repo
+        uses: actions/checkout@v4
+        with:
+          repository: az-digital/az-quickstart-scaffolding
+          path: az-quickstart-scaffolding
+          ref: ${{ env.SCAFFOLD_BRANCH }}
+      - name: Install dependencies
+        run: |
+          cd az-quickstart-scaffolding
+          composer config repositories.az_quickstart vcs https://github.com/az-digital/az_quickstart.git
+          composer config use-github-api false
+          ./quickstart_branch.sh --branch ${AZ_TRIMMED_REF}
+          composer require --no-update --dev az-digital/az-quickstart-dev:dev-${DEV_PACKAGE_BRANCH}
+          composer install -o
+      - name: Save build artifact
+        uses: actions/cache/save@v4
+        with:
+          path: ${{ github.workspace }}/az-quickstart-scaffolding
+          key: azqs-build-${{ github.sha }}
+
+  phpcs:
+    name: PHP_CodeSniffer
+    needs: composer
+    runs-on: ubuntu-latest
+    steps:
+      - uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.3'
+          tools: cs2pr, phpcs
+      - uses: actions/cache/restore@v4
+        with:
+          path: ${{ github.workspace }}/az-quickstart-scaffolding
+          key: azqs-build-${{ github.sha }}
+      - name: Generate phpcs report
+        continue-on-error: true
+        run: |
+          cd $GITHUB_WORKSPACE/az-quickstart-scaffolding
+          ./vendor/bin/phpcs --standard=web/profiles/custom/az_quickstart/phpcs.xml.dist -q --report=checkstyle | sed 's/web\/profiles\/custom\/az_quickstart\///' | cs2pr
+          - name: Run phpcs
+      - name: Run phpcs
+        run: |
+          cd $GITHUB_WORKSPACE/az-quickstart-scaffolding
+          ./vendor/bin/phpcs -sp --colors --standard=web/profiles/custom/az_quickstart/phpcs.xml.dist
+
+  phpstan:
+    name: PHPStan
+    needs: composer
+    runs-on: ubuntu-latest
+    steps:
+      - uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.3'
+          tools: cs2pr, phpstan
+      - uses: actions/cache/restore@v4
+        with:
+          path: ${{ github.workspace }}/az-quickstart-scaffolding
+          key: azqs-build-${{ github.sha }}
+      - name: Generate phpstan report
+        continue-on-error: true
+        run: |
+          cd $GITHUB_WORKSPACE/az-quickstart-scaffolding
+          ./vendor/bin/phpstan analyse --configuration web/profiles/custom/az_quickstart/phpstan.neon web/profiles/custom/az_quickstart --error-format=checkstyle | sed 's/web\/profiles\/custom\/az_quickstart\///' | cs2pr
+      - name: Run phpstan
+        run: |
+          cd $GITHUB_WORKSPACE/az-quickstart-scaffolding
+          ./vendor/bin/phpstan analyse --configuration web/profiles/custom/az_quickstart/phpstan.neon web/profiles/custom/az_quickstart
+
```

Comment:
The PHPStan report is run multiple times here.

Reply:
One provides CLI output and one generates a report for GitHub actions annotations.